### PR TITLE
CLOUDP-227494: internal/pointer,stringutil: add helpers

### PR DESCRIPTION
**Pull request by s-urbaniak**
### All Submissions:

* [ ] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [ ] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if there is one).
* [ ] Update docs/release-notes/release-notes.md if your changes should be included in the release notes for the next release.

### CODE REVIEW

**Comment by s-urbaniak** [comment]
until https://github.com/mongodb/atlas-sdk-go/pull/269 materializes in the SDK, let's use the helpers here for now.

File: internal/pointer/pointer_test.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	t.Run(tc.name, func(t *testing.T) {
+		ptr := SetOrNil(tc.val, tc.defaultValue)
+		if gotNil := ptr == nil; gotNil != tc.wantNil {
+			t.Errorf("got nil %t, want %t", gotNil, tc.wantNil)
+		}
+	})
+}
+
+func TestSetOrNil(t *testing.T) {
+	assertSetOrNil(t, testCase[int]{
```
Comment (josvazg) [inline internal/pointer/pointer_test.go:24]: nit pick, why not putting those in a table, use a loop and use the name as a string for a **t.Run**

Comment (s-urbaniak) [reply internal/pointer/pointer_test.go:24]: because it uses generics which I cannot table test for different types

**Review by josvazg — APPROVED** [review]
Thanks!
LGTM!

**Pull request merged**